Index: /@npmcli/arborist/lib/node.js
===================================================================
--- /@npmcli/arborist/lib/node.js
+++ /@npmcli/arborist/lib/node.js
@@ -1156,9 +1156,15 @@
     }
 
     // if they're links, they match if the targets match
     if (this.isLink) {
-      return node.isLink && this.target.matches(node.target)
+      if (node.isLink) {
+        if (this.target && node.target) {
+          return this.target.matches(node.target)
+        }
+      } else {
+        return false
+      }
     }
 
     // if they're two project root nodes, they're different if the paths differ
     if (this.isProjectRoot && node.isProjectRoot) {
