Index: /@rollup/plugin-commonjs/dist/cjs/index.js
===================================================================
--- /@rollup/plugin-commonjs/dist/cjs/index.js
+++ /@rollup/plugin-commonjs/dist/cjs/index.js
@@ -377,10 +377,11 @@
 
 export function getAugmentedNamespace(n) {
   if (Object.prototype.hasOwnProperty.call(n, '__esModule')) return n;
   var f = n.default;
+  var a
 	if (typeof f == "function") {
-		var a = function a () {
+		a = function a () {
 			if (this instanceof a) {
         return Reflect.construct(f, arguments, this.constructor);
 			}
 			return f.apply(this, arguments);
Index: /@rollup/plugin-commonjs/dist/es/index.js
===================================================================
--- /@rollup/plugin-commonjs/dist/es/index.js
+++ /@rollup/plugin-commonjs/dist/es/index.js
@@ -373,10 +373,11 @@
 
 export function getAugmentedNamespace(n) {
   if (Object.prototype.hasOwnProperty.call(n, '__esModule')) return n;
   var f = n.default;
+  var a
 	if (typeof f == "function") {
-		var a = function a () {
+		a = function a () {
 			if (this instanceof a) {
         return Reflect.construct(f, arguments, this.constructor);
 			}
 			return f.apply(this, arguments);
