Index: /@rollup/plugin-commonjs/dist/cjs/index.js
===================================================================
--- /@rollup/plugin-commonjs/dist/cjs/index.js
+++ /@rollup/plugin-commonjs/dist/cjs/index.js
@@ -377,10 +377,11 @@
 
 export function getAugmentedNamespace(n) {
   if (Object.prototype.hasOwnProperty.call(n, '__esModule')) return n;
   var f = n.default;
+  var a;
 	if (typeof f == "function") {
-		var a = function a () {
+		a = function a () {
 			var isInstance = false;
       try {
         isInstance = this instanceof a;
       } catch {}
Index: /@rollup/plugin-commonjs/dist/es/index.js
===================================================================
--- /@rollup/plugin-commonjs/dist/es/index.js
+++ /@rollup/plugin-commonjs/dist/es/index.js
@@ -373,10 +373,11 @@
 
 export function getAugmentedNamespace(n) {
   if (Object.prototype.hasOwnProperty.call(n, '__esModule')) return n;
   var f = n.default;
+  var a;
 	if (typeof f == "function") {
-		var a = function a () {
+		a = function a () {
 			var isInstance = false;
       try {
         isInstance = this instanceof a;
       } catch {}
