diff --git a/dist/external/debug.js b/dist/external/debug.js
index 2633cad5c0e9105b567fc84bdd9fc354473937e1..7d735a9da9eecc8b7a31285486cf867abfd38a0c 100644
--- a/dist/external/debug.js
+++ b/dist/external/debug.js
@@ -24,6 +24,8 @@ var require_debug = __commonJS({
     debug.names = [];
     debug.skips = [];
     debug.formatters = {};
+    debug.enable = function() {};
+    debug.disable = function() { return ''; };
     module2.exports = debug;
   }
 });
