diff --git a/lib/server-utils.js b/lib/server-utils.js
index a9174052e33039d24b3d153b3850da41a1453255..b07fe0debec568920e3d3d793b3e05e8b5f2c72b 100644
--- a/lib/server-utils.js
+++ b/lib/server-utils.js
@@ -1,7 +1,4 @@
 'use strict';
-var url = require('url')
-  , contrib = require('../index')
-  , blessed = require('blessed');
 
 function OutputBuffer(options) {
   this.isTTY = true;
@@ -41,6 +38,7 @@ function serverError(req, res, err) {
 
 
 function createScreen(req, res) {
+  var url = require('url');
   var query = url.parse(req.url, true).query;
 
   var cols = query.cols || 250;
@@ -53,15 +51,18 @@ function createScreen(req, res) {
 
   res.writeHead(200, {'Content-Type': 'text/plain'});
 
+  var contrib = require('../index')
   var output = new contrib.OutputBuffer({res: res, cols: cols, rows: rows});
   var input = new contrib.InputBuffer(); //required to run under forever since it replaces stdin to non-tty
-  var program = blessed.program({output: output, input: input});
+  var Program = require('blessed/lib/program')
+  var program = new Program({output: output, input: input});
 
   if (query.terminal) program.terminal = query.terminal;
   if (query.isOSX) program.isOSXTerm = query.isOSX;
   if (query.isiTerm2) program.isiTerm2 = query.isiTerm2;
 
-  var screen = blessed.screen({program: program});
+  var ScreenWidget = require('blessed/lib/widgets/screen')
+  var screen = new ScreenWidget({program: program});
   return screen;
 }
 
diff --git a/lib/widget/canvas.js b/lib/widget/canvas.js
index efd10bf4658588afed3580e9fbc897e5a3c69605..17372d84d9817ec8bd528a26cdf2c0d80da4f37b 100644
--- a/lib/widget/canvas.js
+++ b/lib/widget/canvas.js
@@ -1,8 +1,7 @@
 'use strict';
-var blessed = require('blessed')
-  , Node = blessed.Node
-  , Box = blessed.Box
-  , InnerCanvas = require('drawille-canvas-blessed-contrib').Canvas;
+var Box = require('blessed/lib/widgets/box')
+  , InnerCanvas = require('drawille-canvas-blessed-contrib').Canvas
+  , Node = require('blessed/lib/widgets/node');
 
 function Canvas(options, canvasType) {
 
diff --git a/lib/widget/charts/bar.js b/lib/widget/charts/bar.js
index 4d85d902442e01fbd2177f690e61580a31d57ec8..0af052e0b3224e5a7178529e03c948ceb88a3fa5 100755
--- a/lib/widget/charts/bar.js
+++ b/lib/widget/charts/bar.js
@@ -1,6 +1,5 @@
 'use strict';
-var blessed = require('blessed')
-  , Node = blessed.Node
+var Node = require('blessed/lib/widgets/node')
   , Canvas = require('../canvas');
 
 function Bar(options) {
diff --git a/lib/widget/charts/line.js b/lib/widget/charts/line.js
index bd1366eac9f84f2ef4988498deadb324f79e57a5..5f7bba3a4a1b9623735869056e9639c20d0e64f6 100644
--- a/lib/widget/charts/line.js
+++ b/lib/widget/charts/line.js
@@ -1,6 +1,6 @@
 'use strict';
-var blessed = require('blessed')
-  , Node = blessed.Node
+var Box = require('blessed/lib/widgets/box')
+  , Node = require('blessed/lib/widgets/node')
   , Canvas = require('../canvas')
   , utils = require('../../utils.js')
   , _ = require('lodash');
@@ -54,7 +54,7 @@ Line.prototype.setData = function(data) {
     if (!self.options.showLegend) return;
     if (self.legend) self.remove(self.legend);
     var legendWidth = self.options.legend.width || 15;
-    self.legend = blessed.box({
+    self.legend = new Box({
       height: data.length+2,
       top: 1,
       width: legendWidth,
diff --git a/lib/widget/charts/stacked-bar.js b/lib/widget/charts/stacked-bar.js
index ab5917bfe110892b723c3239dcce15bb8c13e2ef..5e9067f2f461042ba6da421fbab1e1ff400596cc 100755
--- a/lib/widget/charts/stacked-bar.js
+++ b/lib/widget/charts/stacked-bar.js
@@ -1,6 +1,6 @@
 'use strict';
-var blessed = require('blessed')
-  , Node = blessed.Node
+var Box = require('blessed/lib/widgets/box')
+  , Node = require('blessed/lib/widgets/node')
   , Canvas = require('../canvas')
   , utils = require('../../utils.js');
 
@@ -184,7 +184,7 @@ StackedBar.prototype.addLegend = function(bars, x) {
   if (!self.options.showLegend) return;
   if (self.legend) self.remove(self.legend);
   var legendWidth = self.options.legend.width || 15;
-  self.legend = blessed.box({
+  self.legend = new Box({
     height: bars.stackedCategory.length+2,
     top: 1,
     width: legendWidth,
diff --git a/lib/widget/donut.js b/lib/widget/donut.js
index ef2dd7fd26b93a1c38e6a74f5d4ced8093793f4c..5ef4fc8c08e9bc8efe0b2197cc9dad28b6ed6b22 100644
--- a/lib/widget/donut.js
+++ b/lib/widget/donut.js
@@ -1,6 +1,5 @@
 'use strict';
-var blessed = require('blessed')
-  , Node = blessed.Node
+var Node = require('blessed/lib/widgets/node')
   , Canvas = require('./canvas');
 
 function Donut(options) {
diff --git a/lib/widget/gauge-list.js b/lib/widget/gauge-list.js
index 05502f0d9897f17ea984b7d12230b8ecad9f7379..281c92b49645074c376a662222f9a5f4af4fbcb4 100644
--- a/lib/widget/gauge-list.js
+++ b/lib/widget/gauge-list.js
@@ -1,6 +1,5 @@
 'use strict';
-var blessed = require('blessed')
-  , Node = blessed.Node
+var Node = require('blessed/lib/widgets/node')
   , Canvas = require('./canvas');
 
 function GaugeList(options) {
diff --git a/lib/widget/gauge.js b/lib/widget/gauge.js
index 228cb85acb06f5fd09856ebfd558414d5fb12844..9a6b37f20e6be919e0c77ab76aa7f6c5bb8e4cf6 100644
--- a/lib/widget/gauge.js
+++ b/lib/widget/gauge.js
@@ -1,6 +1,5 @@
 'use strict';
-var blessed = require('blessed')
-  , Node = blessed.Node
+var Node = require('blessed/lib/widgets/node')
   , Canvas = require('./canvas');
 
 function Gauge(options) {
diff --git a/lib/widget/lcd.js b/lib/widget/lcd.js
index de96d5fa2013d933d7600a6f217196b5e2de33e1..fb302e91bbfc42de31e6064199326c3abecc74cd 100644
--- a/lib/widget/lcd.js
+++ b/lib/widget/lcd.js
@@ -1,6 +1,5 @@
 'use strict';
-var blessed = require('blessed')
-  , Node = blessed.Node
+var Node = require('blessed/lib/widgets/node')
   , Canvas = require('./canvas');
 
 function LCD(options) {
diff --git a/lib/widget/log.js b/lib/widget/log.js
index 86b19d9df717a86d4cc7073cf3d5841aa369e5da..bdb639d7851da58a5660575a05560c5cbd0c56d3 100644
--- a/lib/widget/log.js
+++ b/lib/widget/log.js
@@ -1,7 +1,6 @@
 'use strict';
-var blessed = require('blessed')
-  , Node = blessed.Node
-  , List = blessed.List;
+var List = require('blessed/lib/widgets/list')
+  , Node = require('blessed/lib/widgets/node');
 
 function Log(options) {
   if (!(this instanceof Node)) {
diff --git a/lib/widget/map.js b/lib/widget/map.js
index b14fb755985ceff29179135fb88601de6ecd675d..e1b0116efd2cd7e7566a6a6735c0306db786ac30 100644
--- a/lib/widget/map.js
+++ b/lib/widget/map.js
@@ -1,8 +1,7 @@
 'use strict';
-var blessed = require('blessed')
-  , Node = blessed.Node
-  , Canvas = require('./canvas')
-  , InnerMap = require('map-canvas');
+var InnerMap = require('map-canvas')
+  , Node = require('blessed/lib/widgets/node')
+  , Canvas = require('./canvas');
 
 function Map(options) {
   var self = this;
diff --git a/lib/widget/markdown.js b/lib/widget/markdown.js
index d7f0871423e36fc8dd3a0436c8f3c69f1b3bc299..28ec0e38ac85900abb4503ed80342d30e36cfb47 100644
--- a/lib/widget/markdown.js
+++ b/lib/widget/markdown.js
@@ -1,6 +1,5 @@
 'use strict';
-var blessed = require('blessed')
-  , Box = blessed.Box
+var Box = require('blessed/lib/widgets/box')
   , marked = require('marked')
   , TerminalRenderer = require('marked-terminal')
   , chalk = require('chalk');
diff --git a/lib/widget/picture.js b/lib/widget/picture.js
index c954212f98bce5778bd1c909db9259d5764ef610..9682e039edb9fad33984af430b3d26396b3c4523 100644
--- a/lib/widget/picture.js
+++ b/lib/widget/picture.js
@@ -1,7 +1,6 @@
 'use strict';
-var blessed = require('blessed')
-  , Node = blessed.Node
-  , Box = blessed.Box
+var Box = require('blessed/lib/widgets/box')
+  , Node = require('blessed/lib/widgets/node')
   , pictureTube = require('picture-tuber')
   , fs = require('fs')
   , streams = require('memory-streams')
diff --git a/lib/widget/sparkline.js b/lib/widget/sparkline.js
index 9c0c037934ebc22d8774adaa448e9f37b9e890a4..f7449683f6a19963f8397a3a614be731616c1f0c 100644
--- a/lib/widget/sparkline.js
+++ b/lib/widget/sparkline.js
@@ -1,7 +1,6 @@
 'use strict';
-var blessed = require('blessed')
-  , Node = blessed.Node
-  , Box = blessed.Box
+var Box = require('blessed/lib/widgets/box')
+  , Node = require('blessed/lib/widgets/node')
   , sparkline = require('sparkline');
 
 function Sparkline(options) {
diff --git a/lib/widget/table.js b/lib/widget/table.js
index af253aee50c914f7e66306168fb374106a262feb..a1267e769d669179ac1ab99ddb3560739632a4f2 100644
--- a/lib/widget/table.js
+++ b/lib/widget/table.js
@@ -1,7 +1,7 @@
 'use strict';
-var blessed = require('blessed')
-  , Node = blessed.Node
-  , Box = blessed.Box
+var Box = require('blessed/lib/widgets/box')
+  , List = require('blessed/lib/widgets/list')
+  , Node = require('blessed/lib/widgets/node')
   , stripAnsi = require('strip-ansi');
 
 function Table(options) {
@@ -33,7 +33,7 @@ function Table(options) {
   this.options = options;
   Box.call(this, options);
 
-  this.rows = blessed.list({
+  this.rows = new List({
     //height: 0,
     top: 2,
     width: 0,
diff --git a/lib/widget/tree.js b/lib/widget/tree.js
index c97feb305445a95895fe4a5415a5b55c639432c5..635aa6747e83c8fefe4fdb89b3e79a7d8d17423d 100644
--- a/lib/widget/tree.js
+++ b/lib/widget/tree.js
@@ -1,7 +1,7 @@
 'use strict';
-var blessed = require('blessed'),
-  Node = blessed.Node,
-  Box = blessed.Box;
+var Box = require('blessed/lib/widgets/box')
+  , List = require('blessed/lib/widgets/list')
+  , Node = require('blessed/lib/widgets/node');
 
 function Tree(options) {
   if (!(this instanceof Node)) {
@@ -26,7 +26,7 @@ function Tree(options) {
   options.template.lines = options.template.lines || false;
 
   // Do not set height, since this create a bug where the first line is not always displayed
-  this.rows = blessed.list({
+  this.rows = new List({
     top: 1,
     width: 0,
     left: 1,
