--- a/src/node_contextify.cc
+++ b/src/node_contextify.cc
@@ -262,7 +262,7 @@ MaybeLocal<Context> ContextifyContext::CreateV8Context(
         {},
         {},
         {},
-        {OP(HarmonyDynamicImport), V(kAllowHarmonyDynamicImport, 1)},
+        {OP(HarmonyDynamicImport), V(kAllowHarmonyDynamicImport, 0)},
         {OP(ImportMeta), V(kAllowHarmonyImportMeta, 1)},
         {OP(ImportMetaResolve), V(kAllowHarmonyImportMetaResolve, 1)},
         {}